wagtail_streamfield_migration_toolkit/test/tests/test_migrations.py

```diff
@@ -0,0 +1,258 @@
+                    instance.revisions.all().order_by("id")
+                )
+
+    def apply_migration(self, revisions_from=None):
+        migration = Migration(
+            "test_migration", "wagtail_streamfield_migration_toolkit_test"
+        )
+        migration_operation = MigrateStreamData(
+            app_name="wagtail_streamfield_migration_toolkit_test",
+            model_name=self.model.__name__,
+            field_name="content",
+            operations_and_block_paths=[
+                (
+                    RenameStreamChildrenOperation(
+                        old_name="char1", new_name="renamed1"
+                    ),
+                    "",
+                )
+            ],
+            revisions_from=revisions_from,
+        )
+        migration.operations = [migration_operation]
+
+        loader = MigrationLoader(connection=connection)
+        loader.build_graph()
+        project_state = loader.project_state()
+        schema_editor = connection.schema_editor(atomic=migration.atomic)
+        migration.apply(project_state, schema_editor)
+
+    def _test_migrate_stream_data(self):
+        """Test whether the stream data of the model instances have been updated properly
+
+        Apply the migration and then query the raw data of the updated instances. Compare with
+        original raw data and check whether all relevant `char1` blocks have been renamed and
+        whether ids and other block types are intact.
+        """
+
+        self.apply_migration()
+
+        instances = self.model.objects.all().annotate(
+            raw_content=Cast(F("content"), JSONField())
+        )
+
+        for instance in instances:
+            prev_content = self.original_raw_data[instance.id]
+            for old_block, new_block in zip(prev_content, instance.raw_content):
+                self.assertEqual(old_block["id"], new_block["id"])
+                if old_block["type"] == "char1":
+                    self.assertEqual(new_block["type"], "renamed1")
+                else:
+                    self.assertEqual(old_block["type"], new_block["type"])
+
+    def _test_migrate_revisions(self):
+        """Test whether all revisions have been updated properly
+
+        Applying migration with `revisions_from=None`, so all revisions should be updated.
+        """
+
+        self.apply_migration()
+
+        instances = self.model.objects.all().annotate(
+            raw_content=Cast(F("content"), JSONField())
+        )
+
+        for instance in instances:
+            old_revisions = self.original_revisions[instance.id]
+            for old_revision, new_revision in zip(
+                old_revisions, instance.revisions.all().order_by("id")
+            ):
+                old_content = json.loads(old_revision.content["content"])
+                new_content = json.loads(new_revision.content["content"])
+                for old_block, new_block in zip(old_content, new_content):
```

Comment:
You still repeat this loop over old blocks and new blocks, asserting they're identical many times. Could this become its own method - eg `def assertBlocksRenamed(old_content, new_content)` that handles the sub-assertions itself and avoids you rewriting this loop of assertions several times?

Reply:
Ah right, I thought you were referring to something else earlier :sweat_smile: